### ISSUE TITLE
VisibleExtent recalc after base map load

### ISSUE THREAD

**Issue by katterfelto** [wfletcher/EasyGIS.NET#85]
I am having a problem with the VisibleExtent property after I load a base map TileSource. 

If I load [ngr.csv](https://github.com/wfletcher/EasyGIS.NET/files/9506991/ngr.csv) into the Easy GIS Desktop app, it zooms to extents and report a reasonable width of 8331m on the status bar.

After I load the Open Street Map. I need to resize the window to get my width extent about the same, but the width on the status bar is now reported as 13300m. **Any pointers to how I can fix that would be appreciated.** 

Just an aside if I load the lat/long version of the file [latlong.csv](https://github.com/wfletcher/EasyGIS.NET/files/9506994/latlong.csv), the most easterly point (The Royal Observatory) which is very close to the Meridian is not plotted when the base map is displayed.

**Comment by wfletcher** [comment]
I think that the status bar width uses the maps visible extent, calculating the width if the CRS is WGS84 lat/Lon using distance formulas. If the map CRS is a projected CRS then the label  width is calculated using VisibleExtent.Right -VisibleExtent.Left.

 I suspect that there may be a bug in there somewhere and will look into it

**Comment by wfletcher** [comment]
I have looked into this. 

Currently I use ellipsoidal calculations for the visible extent and measurements (control/shift key + mouse left/right click and drag) if the map is using a "geographic" CRS, such as WGS84(EPSG:4326).  If the map is using a projected CRS then I use Cartesian calculations. 

I am assuming that the "ngv.csv" is using the "British National Grid" CRS. In this case you will find that measurements will be almost identical if you change the map CRS to WGS84. The issue you are having when you load open street map is that the map switches the CRS to "WGS84/Pseudo-Mercator", which has terrible distortion for locations far from the equator.

The solution is to always use ellipsoidal calculations for measurements (even when using a projected CRS) or at least have an option to specify whether to use cartesian or ellipsoidal. This is similar to what QGIS uses in their measurement tool.

I will create an issue to provide an option to specify whether measurements should use cartesian or ellipsoidal calculations.

I have also looked into the issue where the most easterly point is not displayed and can confirm that this is a bug
